graph/layout/optimizer.go

```diff
@@ -0,0 +1,106 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package layout
+
+import (
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/spatial/r2"
+)
+
+// LayoutR2 implements graph layout updates and representations.
+type LayoutR2 interface {
+	// IsInitialized returns whether the Layout is initialized.
+	IsInitialized() bool
```

Comment:
Why the need for this?

Reply:
This allows the updater to know whether layout needs to be started. Without it we need to perform the intialisation prior to the loop.